### ISSUE TITLE
[tune] Add documentation, tests, and example for fractional GPU usage

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#2846]
### Describe the problem
Followup to #2768 for public documentation (characterized as experimental).

**Comment by richardliaw** [comment]
May be broken actually: Running the following in rllib.

```
atari-a2c:
    env:
        grid_search:
            - BreakoutNoFrameskip-v4
            - BeamRiderNoFrameskip-v4
            - SeaquestNoFrameskip-v4
            - QbertNoFrameskip-v4
    run: A2C
    stop: {
        "timesteps_total": 5000000
    }
    checkpoint_freq: 50
    local_dir: "/data/rliaw/rl_experiments/a2c/"
    config:
        sample_batch_size: 100
        num_workers: 5
        use_gpu_for_workers: false
        # num_envs_per_worker: 5
        gpu: true
        gpu_fraction: 0.25
```

Leads me with this output:
```
== Status ==
Using FIFO scheduling algorithm.


Created LogSyncer for /data/rliaw/rl_experiments/a2c/atari-a2c/A2C_BreakoutNoFrameskip-v4_0_env=BreakoutNoFrameskip-v4_2018-09-09_23-56-45sdx3__zx ->
== Status ==
Using FIFO scheduling algorithm.
Resources requested: 6/48 CPUs, 0.25/1 GPUs
Result logdir: /data/rliaw/rl_experiments/a2c/atari-a2c
PENDING trials:
 - A2C_BeamRiderNoFrameskip-v4_1_env=BeamRiderNoFrameskip-v4:   PENDING
 - A2C_SeaquestNoFrameskip-v4_2_env=SeaquestNoFrameskip-v4:     PENDING
 - A2C_QbertNoFrameskip-v4_3_env=QbertNoFrameskip-v4:   PENDING
RUNNING trials:
 - A2C_BreakoutNoFrameskip-v4_0_env=BreakoutNoFrameskip-v4:     RUNNING

Created LogSyncer for /data/rliaw/rl_experiments/a2c/atari-a2c/A2C_BeamRiderNoFrameskip-v4_1_env=BeamRiderNoFrameskip-v4_2018-09-09_23-56-45lu9iqh2w ->
Created LogSyncer for /data/rliaw/rl_experiments/a2c/atari-a2c/A2C_SeaquestNoFrameskip-v4_2_env=SeaquestNoFrameskip-v4_2018-09-09_23-56-45ppx7p5b2 ->
Created LogSyncer for /data/rliaw/rl_experiments/a2c/atari-a2c/A2C_QbertNoFrameskip-v4_3_env=QbertNoFrameskip-v4_2018-09-09_23-56-45yob4nv7n ->
F0909 23:56:47.516850 14586 local_scheduler.cc:447]  Check failed: state->available_gpus.size() >= resource_quantity
*** Check failure stack trace: ***
    @           0x4a59b6  google::LogMessage::Fail()
    @           0x4a58fa  google::LogMessage::SendToLog()
    @           0x4a523c  google::LogMessage::Flush()
    @           0x4a5037  google::LogMessage::~LogMessage()
    @           0x494f00  ray::RayLog::~RayLog()
    @           0x42e3c8  acquire_resources()
    @           0x432581  assign_task_to_worker()
    @           0x43d711  dispatch_tasks()
    @           0x43d863  dispatch_all_tasks()
    @           0x43f8f2  handle_worker_available()
    @           0x435d00  process_message()
    @           0x46dd48  aeProcessEvents
    @           0x46e0eb  aeMain
    @           0x42b415  main
    @     0x7f4efb0fb830  __libc_start_main
    @           0x42c1f9  _start
    @              (nil)  (unknown)
```

**Comment by richardliaw** [comment]
Closed by #3117
